dingo/gurobi_based_implementations.py

```diff
@@ -273,6 +274,601 @@ def fast_fva(lb, ub, S, c, opt_percentage=100):
+                        model_iter,
+                        n,
+                        Aeq_sparse,
+                        beq,
+                        lb,
+                        ub_iter,
+                        objective_function_max,
+                    )
+                    model_iter.optimize()
+
+                    status = model_iter.status
+                    if status == GRB.OPTIMAL:
+                        # Get the max objective value
+                        max_objective2 = -model_iter.getObjective().getValue()
+                        print(np.abs(max_objective2 - max_objective))
+                        if np.abs(max_objective2 - max_objective) > red_facet_tol:
+                            redundant_facet_right = False
+
+                    model_iter = update_model(
+                        model_iter, n, Aeq_sparse, beq, lb, ub, objective_function
+                    )
+                    model_iter.optimize()
+
+                    # If optimized
+                    status = model_iter.status
+                    if status == GRB.OPTIMAL:
+                        # Get the min objective value
+                        min_objective = model_iter.getObjective().getValue()
+                    else:
+                        min_objective = lb[i]
+
+                    lb_iter = lb
+                    lb_iter[i] = lb[i] - 1
+                    model_iter = update_model(
+                        model_iter, n, Aeq_sparse, beq, lb_iter, ub, objective_function
+                    )
+                    model_iter.optimize()
+
+                    status = model_iter.status
+                    if status == GRB.OPTIMAL:
+                        # Get the max objective value
+                        min_objective2 = model_iter.getObjective().getValue()
+                        print(np.abs(min_objective2 - min_objective))
+                        if np.abs(min_objective2 - min_objective) > red_facet_tol:
+                            redundant_facet_left = False
+
+                    if (not redundant_facet_left) or (not redundant_facet_right):
+                        width = abs(max_objective - min_objective)
+                        print(width)
+
+                        # Check whether we keep or not the equality
+                        if width < red_facet_tol:
+                            Aeq_res = np.vstack(
+                                (
+                                    Aeq_res,
+                                    A[
+                                        i,
+                                    ],
+                                )
+                            )
+                            beq_res = np.append(
+                                beq_res, min(max_objective, min_objective)
+                            )
+                        else:  # fix this
+                            if not redundant_facet_left:
+                                A_res = np.append(
```

Comment:
why not writing this in one line instead of 10 ?

Reply:
This comes from black package